File: src/main/java/org/hwabeag/playerfly/events/JoinEvent.java
package org.hwabeag.playerfly.events;

import java.util.Objects;
import org.bukkit.GameMode;
import org.bukkit.configuration.file.FileConfiguration;
import org.bukkit.entity.Player;
import org.bukkit.event.EventHandler;
import org.bukkit.event.Listener;
import org.bukkit.event.player.PlayerJoinEvent;
import org.hwabeag.playerfly.config.ConfigManager;

public class JoinEvent implements Listener {
    FileConfiguration PlayerFlyConfig = ConfigManager.getConfig("playerfly");

    public JoinEvent() {
    }

    @EventHandler
    public void onPlayerJoin(PlayerJoinEvent event) {
        Player player = event.getPlayer();
        String name = ((Player)Objects.requireNonNull(player)).getName();
        if (this.PlayerFlyConfig.getString("플라이." + name + ".남은시간") == null) {
            this.PlayerFlyConfig.addDefault("플라이." + name + ".남은시간", 0);
            this.PlayerFlyConfig.addDefault("플라이." + name + ".관리", "오프");
            ConfigManager.saveConfigs();
        }

        if (player.getGameMode() != GameMode.CREATIVE) {
            player.setFlying(false);
            player.setAllowFlight(false);
            this.PlayerFlyConfig.set("플라이." + name + ".관리", "오프");
            ConfigManager.saveConfigs();
        }

    }
}

File: src/main/kotlin/org/hwabeag/playerfly/events/JoinEvent.kt
package org.hwabeag.playerfly.events

import org.bukkit.GameMode
import org.bukkit.event.EventHandler
import org.bukkit.event.Listener
import org.bukkit.event.player.PlayerJoinEvent
import org.hwabeag.playerfly.PlayerFlyPlugin

class JoinEvent(private val plugin: PlayerFlyPlugin) : Listener {
    @EventHandler
    fun onJoin(event: PlayerJoinEvent) {
        val player = event.player
        val data = plugin.databaseManager.getData(player.uniqueId, player.name)

        // 세션 시작 시 플라이 활성 상태는 초기화하되, 타 플러그인의 비행 권한은 건드리지 않는다.
        if (data.enabled) {
            plugin.databaseManager.setEnabled(player.uniqueId, player.name, false)
        }

        if (player.gameMode != GameMode.CREATIVE && player.gameMode != GameMode.SPECTATOR && player.isFlying) {
            player.isFlying = false
        }
        plugin.unmarkManagedFlight(player.uniqueId)
    }
}
